src/tls/keypair.h

```diff
@@ -54,10 +54,47 @@ namespace tls
 #elif CURVE_CHOICE_SECP256K1_MBEDTLS || CURVE_CHOICE_SECP256K1_BITCOIN
   static constexpr mbedtls_md_type_t MD_TYPE = MBEDTLS_MD_SHA256;
   static constexpr size_t MD_SIZE = 256 / 8;
+  static constexpr size_t REC_ID_IDX = 64;
 #endif
 
   using Hash = std::array<uint8_t, MD_SIZE>;
 
+#if CURVE_CHOICE_SECP256K1_BITCOIN
+  inline bool verify_secp256k_bc(
+    secp256k1_context* ctx, const uint8_t* signature, const uint8_t* hash)
+  {
+    secp256k1_pubkey public_key;
+    secp256k1_ecdsa_recoverable_signature sig;
+
+    if (
+      secp256k1_ecdsa_recoverable_signature_parse_compact(
+        ctx, &sig, signature, signature[REC_ID_IDX]) != 1)
+    {
+      LOG_INFO_FMT(
```

Comment:
Here and in error cases in this function: `LOG_FAIL_FMT`

Reply:
Not sure this should be logged as an error. If we get a lot of invalid signed requests, it'll spam our logs, and we never really logged these failures before.

Reply:
True. Perhaps `LOG_DEBUG_FMT` then? info will spam the logs just as much as fail in release builds.